components/raftstore-v2/src/router/message.rs

```diff
@@ -0,0 +1,293 @@
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(u8)]
+pub enum PeerTick {
+    Raft = 0,
+    RaftLogGc = 1,
+    SplitRegionCheck = 2,
+    PdHeartbeat = 3,
+    CheckMerge = 4,
+    CheckPeerStaleState = 5,
+    EntryCacheEvict = 6,
+    CheckLeaderLease = 7,
+    ReactivateMemoryLock = 8,
+    ReportBuckets = 9,
+}
+
+impl PeerTick {
+    pub const VARIANT_COUNT: usize = Self::get_all_ticks().len();
+
+    #[inline]
+    pub fn tag(self) -> &'static str {
+        match self {
+            PeerTick::Raft => "raft",
+            PeerTick::RaftLogGc => "raft_log_gc",
+            PeerTick::SplitRegionCheck => "split_region_check",
+            PeerTick::PdHeartbeat => "pd_heartbeat",
+            PeerTick::CheckMerge => "check_merge",
+            PeerTick::CheckPeerStaleState => "check_peer_stale_state",
+            PeerTick::EntryCacheEvict => "entry_cache_evict",
+            PeerTick::CheckLeaderLease => "check_leader_lease",
+            PeerTick::ReactivateMemoryLock => "reactivate_memory_lock",
+            PeerTick::ReportBuckets => "report_buckets",
+        }
+    }
+
+    pub const fn get_all_ticks() -> &'static [PeerTick] {
+        const TICKS: &[PeerTick] = &[
+            PeerTick::Raft,
+            PeerTick::RaftLogGc,
+            PeerTick::SplitRegionCheck,
+            PeerTick::PdHeartbeat,
+            PeerTick::CheckMerge,
+            PeerTick::CheckPeerStaleState,
+            PeerTick::EntryCacheEvict,
+            PeerTick::CheckLeaderLease,
+            PeerTick::ReactivateMemoryLock,
+            PeerTick::ReportBuckets,
+        ];
+        TICKS
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum StoreTick {
+    // No CompactLock and CompactCheck as they should be implemented by peer itself.
+    PdStoreHeartbeat,
+    SnapGc,
+    ConsistencyCheck,
+    CleanupImportSst,
+}
+
+impl StoreTick {
+    #[inline]
+    pub fn tag(self) -> RaftEventDurationType {
+        match self {
+            StoreTick::PdStoreHeartbeat => RaftEventDurationType::pd_store_heartbeat,
+            StoreTick::SnapGc => RaftEventDurationType::snap_gc,
+            StoreTick::ConsistencyCheck => RaftEventDurationType::consistency_check,
+            StoreTick::CleanupImportSst => RaftEventDurationType::cleanup_import_sst,
+        }
+    }
+}
+
+/// Raft command is the command that is expected to be proposed by the
+/// leader of the target raft group.
+#[derive(Debug)]
+pub struct RaftCommand<S: Snapshot> {
+    pub send_time: Instant,
+    pub request: RaftCmdRequest,
+    pub ch: ResponseChannel<S>,
+}
+
+impl<S: Snapshot> RaftCommand<S> {
+    #[inline]
+    pub fn new(request: RaftCmdRequest, ch: ResponseChannel<S>) -> RaftCommand<S> {
+        RaftCommand {
+            request,
+            ch,
+            send_time: Instant::now(),
+        }
+    }
+}
+
+/// Message that can be sent to a peer.
+pub enum PeerMsg<EK: KvEngine> {
```

Comment:
Comparing with V1, it does not have CasualMessage, HeartbeatPd, Destroy.   So are they not needed anymore in v2?

Reply:
No, I delete them on purpose and add them as needed in future development.